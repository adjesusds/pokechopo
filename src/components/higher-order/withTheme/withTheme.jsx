// External Packages
import React from 'react'
import { ThemeProvider } from 'styled-components'

// Internal Packages
import theme from 'config/theme'
import getComponentDisplayName from 'utils/getComponentDisplayName'

/**
 * Creates a new withTheme higher-order component.
 * @class
 */
const withTheme = WrappedComponent => {
  const WithTheme = props => (
    <ThemeProvider theme={theme}>
      <WrappedComponent {...props} />
    </ThemeProvider>
  )

  /** @override */
  WithTheme.displayName = `WithTheme(${getComponentDisplayName(WrappedComponent)})`

  return WithTheme
}

export default withTheme
