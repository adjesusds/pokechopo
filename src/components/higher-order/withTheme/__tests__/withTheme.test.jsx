// External Packages
import { shallow } from 'enzyme'
import React from 'react'

// Internal Packages
import withTheme from '../withTheme'

describe('<WithTheme(WrappedComponent) />', () => {
  it('should wrap the component on valid props', () => {
    const WrappedComponent = 'span'

    const WrappedComponentWithTheme = withTheme(WrappedComponent)

    const wrappedComponent = shallow(<WrappedComponentWithTheme>Hello, World!</WrappedComponentWithTheme>)

    expect(wrappedComponent).toMatchSnapshot()
  })
})
