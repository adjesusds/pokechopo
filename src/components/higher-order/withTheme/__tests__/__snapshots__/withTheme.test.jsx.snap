// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithTheme(WrappedComponent) /> should wrap the component on valid props 1`] = `
<ThemeProvider
  theme={
    Object {
      "mode": "normal",
    }
  }
>
  <span>
    Hello, World!
  </span>
</ThemeProvider>
`;
